lib/spack/spack/cmd/develop.py

```diff
@@ -151,4 +153,17 @@ def develop(parser, args):
     env = spack.cmd.require_active_env(cmd_name="develop")
     tty.debug("Updating develop config for {0} transactionally".format(env.name))
     with env.write_transaction():
+        if args.build_directory is not None:
+            scope = env.env_file_config_scope_name()
+            spack.config.add(
+                "packages:{}:package_attributes:build_directory:{}".format(
+                    spec.name, args.build_directory
+                ),
+                scope,
+            )
+            tty.warn(
```

Comment:
@becker33 I added this warning, but I don't think I like it. I imagine teams that rely on this feature will find it very annoying over time.  Do you think the doc's warning is sufficient?

Reply:
I agree we should remove the warning